src/tests/src/solver/modelParser/test_full.cpp

```diff
@@ -0,0 +1,510 @@
+
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <yaml-cpp/exceptions.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/solver/libObjectModel/library.h"
+#include "antares/solver/modelConverter/modelConverter.h"
+#include "antares/solver/modelParser/Library.h"
+#include "antares/solver/modelParser/parser.h"
+
+#include "enum_operators.h"
+
+using namespace std::string_literals;
+using namespace Antares::Solver;
+
+void checkParameter(const ObjectModel::Parameter& parameter,
+                    const std::string& name,
+                    bool timeDependent,
+                    bool scenarioDependent,
+                    ObjectModel::ValueType type)
+{
+    std::cout << "Parameter: " << parameter.Id() << std::endl;
```

Comment:
```suggestion
```

Reply:
Helpful when an error occurs. Else we don't really know what failed.